src/main/webapp/app/exercises/programming/manage/grading/tasks/programming-exercise-grading-tasks-table.component.ts

```diff
@@ -0,0 +1,150 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ProgrammingExerciseTaskService } from 'app/exercises/programming/manage/grading/tasks/programming-exercise-task.service';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { Course } from 'app/entities/course.model';
+import { faAngleDown, faAngleRight, faAsterisk, faMedal, faQuestionCircle, faScaleUnbalanced, faSort, faSortDown, faSortUp } from '@fortawesome/free-solid-svg-icons';
+import { ProgrammingExerciseGradingStatistics } from 'app/entities/programming-exercise-test-case-statistics.model';
+import { ProgrammingExerciseTask } from './programming-exercise-task';
+import { Observable, Subject } from 'rxjs';
+
+type Sort = {
+    by: 'name' | 'weight' | 'multiplier' | 'bonusPoints' | 'visibility' | 'resulting' | 'type';
+    descending: boolean;
+};
+
+type TaskComparator = (a: ProgrammingExerciseTask, b: ProgrammingExerciseTask) => number;
+
+@Component({
+    selector: 'jhi-programming-exercise-grading-tasks-table',
+    templateUrl: './programming-exercise-grading-tasks-table.component.html',
+    styleUrls: ['./programming-exercise-grading-tasks-table.scss'],
+})
+export class ProgrammingExerciseGradingTasksTableComponent implements OnInit {
+    @Input() exercise: ProgrammingExercise;
+    @Input() course: Course;
+    @Input() gradingStatisticsObservable: Observable<ProgrammingExerciseGradingStatistics>;
+
+    // Icons
+    faAngleDown = faAngleDown;
+    faAngleRight = faAngleRight;
+    faQuestionCircle = faQuestionCircle;
+    faScaleUnbalanced = faScaleUnbalanced;
+    faMedal = faMedal;
+    faAsterisk = faAsterisk;
+
+    isSaving = false;
+    tasks: ProgrammingExerciseTask[];
+    allTasksExpandedSubject: Subject<boolean>;
+
+    currentSort: Sort | undefined;
+
+    get ignoreInactive() {
+        return this.taskService.ignoreInactive;
+    }
+
+    constructor(private taskService: ProgrammingExerciseTaskService) {}
+
+    ngOnInit(): void {
+        this.allTasksExpandedSubject = new Subject();
+        this.gradingStatisticsObservable.subscribe((gradingStatistics) => {
+            this.taskService.configure(this.exercise, this.course, gradingStatistics).subscribe(this.updateTasks);
+        });
```

Comment:
Should we unsubscribe both of these subscriptions on ngDestroy to avoid memory leaks or are those finite subscriptions?

Reply:
Both are finite (they stem from http calls) so unsubscribing shouldn't be necessary